server/src/main/java/com/hedera/block/server/persistence/storage/compression/Compression.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.persistence.storage.compression;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Path;
+
+/**
+ * A compression abstractions that allows for the compression of bytes using
+ * different compression algorithms based on specific implementation.
+ */
+public interface Compression {
+    /**
+     * This method aims to create a new {@link OutputStream} instance that will
+     * run the input data through a compression algorithm (algorithm is based on
+     * specific implementation) before writing it to the input path. The input
+     * path must be a file and not a directory. This method and all it`s
+     * implementations are NOT responsible for creating the path to
+     * the parent directory of the input file path provided, but WILL create the
+     * actual file if it does not exist.
```

Comment:
I am still a bit uncertain about this method api contract about not being responsible for creating the path to the parent of the desired file where we want to write the compressed data. Internally, both implementations use `Files#newOutputStream(Path)` which will create the actual file we want to write to if it does not exist, but it will throw an exception if the path to the parent folder of our target path does not exist. Should it be part of the api that the new output stream shall ensure also the creation of the parent of the desired target? My point here is if we can ensure that all possible other implementations can (and should) enforce that logic, since this is the highest level abstraction and all specific implementations **must** comply with this contract. Opinions?

Reply:
Why is Compression creating files?  I would expect the Resolver to have a method for creating a file...
My reasoning is that not all implementations will create files, and having the Resolver do so (optionally, via a method call) seems the most logical place to add that function.

Reply:
Yep. My point exaclty. I do not think that file creation should be the responsibility of the `Compression` abstraction. In fact that was my initial implementation. I think I will revert to that... It also does not seem _appropriate_ that all specific implementations would fit this contract.

Reply:
✅ Done. The compression now wraps a stream instead.